c2rust-analyze/src/context.rs

```diff
@@ -98,13 +98,15 @@ pub struct AnalysisCtxt<'a, 'tcx> {
     /// those `PointerId`s consistent, the `Rvalue`'s type must be stored rather than recomputed on
     /// the fly.
     pub rvalue_tys: HashMap<Location, LTy<'tcx>>,
+    pub const_tys: HashMap<ConstantKind<'tcx>, LTy<'tcx>>,
     next_ptr_id: NextLocalPointerId,
 }
 
 pub struct AnalysisCtxtData<'tcx> {
     local_tys: IndexVec<Local, LTy<'tcx>>,
     addr_of_local: IndexVec<Local, PointerId>,
     rvalue_tys: HashMap<Location, LTy<'tcx>>,
+    const_tys: HashMap<ConstantKind<'tcx>, LTy<'tcx>>,
```

Comment:
It looks to me like we could use `rvalue_tys` for this instead of adding a new map.

Reply:
Discussion moved here: https://github.com/immunant/c2rust/pull/886#issuecomment-1517568518.